### DISCUSSION TITLE
Spam-safe user registration.

### DISCUSSION THREAD

**Discussion by EricSoroos** [ckan/ckan#8119] [Ideas]
Re #8038 -- User spam

There's an issue with user registration and spam, because by default, there is a path for new users to upload content that is served to the public, through the user's profile image, or the user full name/notes when the show users setting is allowed. 

For sites that have a business requirement for open registration, this is a significant administrative load, even after adding captchas. We're seeing on the order of thousands of attempts per day to register and post spam, vs maybe one authentic registration. 

One potential option here would be to reduce the incentive to create users. 

* Remove the profile image
* Definitely make the user list private 
* Potentially require confirmation by admins.

Additionally, saving some extra information to audit the user registration, prior to confirmation (which would need to be cleared periodically):
* IP Address (depending on your stance on GDPR this should be permitted as a technical site functionality requirement)
* User Agent

**Comment by wardi** [comment]
I like these ideas.

How about an option to enable user image uploads (new default `false`)  instead of removing the feature completely?

`ckan.public_user_details` default can be switched to `false`

Account confirmation is great, we've implemented it in our plugin and by forcing all datasets to private on our internal site. Having a user state of not-approved-yet would be even better
